option py_generic_services = true;


message CPUInfo {
    required string name = 1;
    required int32 user_count = 2;
    required int32 nice_count = 3;
    required int32 sys_count = 4;
    required int32 idle_count = 5;
    required int32 iowait_count = 6;
    required int32 total_count = 7;
}

message NetInfo {
    required string name = 1;
    required string ip = 2;
    required int64 recv_byte = 3;
    required int64 send_byte = 4;
}

message VirtualMemInfo {
    required int64 total = 1;
    required int64 available = 2;
    required int64 used = 3;
    required int32 percent = 4;
}

message SwapMemInfo {
    required int64 total = 1;
    required int64 free = 2;
    required int64 used = 3;
    required int32 percent = 4;
}

message MemInfo {
    required VirtualMemInfo virtual_mem = 1;
    required SwapMemInfo swap_mem = 2;
}

message ReportRequest {
    required string host_name = 1;
    required int64 datetime = 2;
    repeated CPUInfo cpu_infos = 3;
    repeated NetInfo net_infos = 4;
    required MemInfo mem_info = 5;
}

message ReportResponse {
    required int32 return_code = 1;
    required string msg = 2;
}

message RegisterRequest {
    required string host_name = 1;
}

message RegisterResponse {
    required int32 return_code = 1;
    required string msg = 2;
}


// Test service
service MonsitService {
  rpc Register (RegisterRequest) returns (RegisterResponse);
  rpc Report (ReportRequest) returns (ReportResponse);
}